### PR TITLE
Limit project name characters

### PR DESCRIPTION
**Pull request by jasmingoh**
This PR addresses #718 ([ADO task](https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/30494))

Users can only enter a project name up to 50 characters in the input field. If their project name is 50 characters long, an error message will show and they will not be able to proceed to the next page:

This is also address the concern that @SahilTara brought up in https://github.com/microsoft/WebTemplateStudio/pull/824#discussion_r300796350

### CODE REVIEW

File: src/client/src/utils/validateName.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { defineMessages, FormattedMessage } from "react-intl";
 
+const PROJECT_NAME_CHARACTER_LIMIT = 50;
```
Comment (SahilTara) [inline src/client/src/utils/validateName.ts]: just curious, why 50?

Comment (jasmingoh) [reply src/client/src/utils/validateName.ts]: Our thought is that developers wouldn't put in such a long project name, and Visual Studio also limit the project name to 50 characters.

Comment (xtinah-w) [reply src/client/src/utils/validateName.ts]: We also based the project name input off of the Windows Template Studio char limit for project name!

Comment (SahilTara) [reply src/client/src/utils/validateName.ts]: Very small nit that comes from this though, can we add this to the constants file, and use the constant here and ProjectNameAndOutput?

**Review by SahilTara — APPROVED** [review]

**Review by jsondoo — APPROVED** [review]

**Review by xtinah-w — APPROVED** [review]
Looks good and it works on my system

**Pull request merged**